library/include/hipsparselt.h

```diff
@@ -39,6 +39,32 @@
 //! cuSPARSELt "version 0.3".
 //
 
+
+/** @defgroup types_module Data types
+ *
+ */
+/** @defgroup library_module Library management functions
+ * Provides the library handle
+ */
+/** @defgroup matrix_desc_module Matrix descriptor functions
+ * Defines sparse and dense matrices
+ */
+/** @defgroup matmul_module Matrix multiply (matmul) functions
+ * Multiplies sparse matrices
+ */
+/** @defgroup matmul_desc_module Matmul descriptor functions
+ * Defines how to matrix multiply
+ */
+/** @defgroup matmul_algo_module Matmul algorithm functions
+ * Provides algorithms for matrix multiply
+ */
+/** @defgroup helper_module Helper functions
+ * Required for subsequent library calls
+ */
+/** @defgroup aux_module Auxilary functions
+ * Initializes hipSPARSELt for the current HIP device
+ */
```

Comment:
I believe there's no need to open/close the documentation comments for each entry. The following would work fine too:

```cpp
/** @defgroup module_a
  * Description of module A.
  * 
  * @defgroup module_b
  * Description of module B.
  */
```

At StreamHPC we've decided to stick with `/// <documentation>` instead of `/** <documentation */, and `\command` over `@command`. I'm currently not aware of an AMD standard regarding Doxygen comments. In our style this would be:

```cpp
/// \defgroup module_a
/// Description of module A.
/// 
/// \defgroup module_b
/// Description of module B.
```

I can send the details of the standard we picked and the decisions for picking between the different options over email, but simply said, there are many different ways and all are fine. It's mostly a matter of having everyone between the different AMD projects agree on the same set of rules.

Reply:
I also noticed that the rest of the file uses Qt-style doxygen comments:
```cpp
/*! \ingroup types_module
 *  ...
 */
```